climada/hazard/centroids/centr.py

```diff
@@ -478,44 +479,44 @@ def select_mask(self, sel_cen=None, reg_id=None, extent=None):
               (self.lat >= lat_min) & (self.lat <= lat_max)
             )
         return sel_cen
-
-    #TODO replace with nice GeoDataFrame util plot method.
-    def plot(self, axis=None, figsize=(9, 13), **kwargs):
-        """Plot centroids scatter points over earth
+    
+    def plot(self, axis=None, figsize=(9, 13), *args, **kwargs):
+        """Plot centroids geodataframe using geopandas and cartopy plotting functions.
 
         Parameters
         ----------
-        axis : matplotlib.axes._subplots.AxesSubplot, optional
-            axis to use
+        axis: optional
+            user-defined cartopy.mpl.geoaxes.GeoAxes instance
         figsize: (float, float), optional
             figure size for plt.subplots
             The default is (9, 13)
+        args : optional
+            positional arguments for geopandas.GeoDataFrame.plot
         kwargs : optional
-            arguments for scatter matplotlib function
-
+            keyword arguments for geopandas.GeoDataFrame.plot
+            
         Returns
         -------
-        axis : matplotlib.axes._subplots.AxesSubplot
+        ax : cartopy.mpl.geoaxes.GeoAxes instance
         """
-        pad = np.abs(u_coord.get_resolution(self.lat, self.lon)).min()
-
-        proj_data, _ = u_plot.get_transformation(self.crs)
-        proj_plot = proj_data
-        if isinstance(proj_data, ccrs.PlateCarree):
-            # use different projections for plot and data to shift the central lon in the plot
-            xmin, ymin, xmax, ymax = u_coord.latlon_bounds(self.lat, self.lon, buffer=pad)
-            proj_plot = ccrs.PlateCarree(central_longitude=0.5 * (xmin + xmax))
-        else:
-            xmin, ymin, xmax, ymax = (self.lon.min() - pad, self.lat.min() - pad,
-                                      self.lon.max() + pad, self.lat.max() + pad)
+        if axis == None:
+            fig, axis = plt.subplots(figsize=figsize, subplot_kw={"projection": ccrs.PlateCarree()})
+        if type(axis) != cartopy.mpl.geoaxes.GeoAxes:
+            raise AttributeError(
+                f"The axis provided is of type: {type(axis)} "
+                "The function requires a cartopy.mpl.geoaxes.GeoAxes."
+            )
 
-        if not axis:
-            _fig, axis, _fontsize = u_plot.make_map(proj=proj_plot, figsize=figsize)
+        axis.add_feature(cfeature.BORDERS)
+        axis.add_feature(cfeature.COASTLINE)
+        axis.gridlines(draw_labels=True, dms=True, x_inline=False, y_inline=False)
 
-        axis.set_extent((xmin, xmax, ymin, ymax), crs=proj_data)
-        u_plot.add_shapes(axis)
-        axis.scatter(self.lon, self.lat, transform=proj_data, **kwargs)
-        plt.tight_layout()
+        if self.gdf.crs != DEF_CRS:
+            copy_gdf = copy.deepcopy(self)
+            copy_gdf.to_default_crs()
+            copy_gdf.gdf.plot(ax=axis, transform=ccrs.PlateCarree(), *args, **kwargs)
```

Comment:
```suggestion
            centroids_plot = to_default_crs(inplace=False)
            centroids_plot.gdf.plot(ax=axis, transform=ccrs.PlateCarree(), *args, **kwargs)
```

Reply:
Wouldn't it need to be `centroids_plot = self.to_default_crs(inplace=False)` ?

Reply:
correct!